services/src/main/java/org/keycloak/protocol/oid4vc/issuance/signing/LDSigningService.java

```diff
@@ -31,16 +44,57 @@
  */
 public class LDSigningService extends SigningService<VerifiableCredential> {
 
+    private final LinkedDataCryptographicSuite linkedDataCryptographicSuite;
+    private final TimeProvider timeProvider;
+    private final String keyId;
 
-    public LDSigningService(KeycloakSession keycloakSession, String keyId, String ldpType) {
-        super(keycloakSession, keyId, ldpType);
+    public LDSigningService(KeycloakSession keycloakSession, String keyId, String algorithmType, String ldpType, ObjectMapper objectMapper, TimeProvider timeProvider, Optional<String> kid) {
+        super(keycloakSession, keyId, algorithmType);
+        this.timeProvider = timeProvider;
+        this.keyId = kid.orElse(keyId);
+        KeyWrapper signingKey = getKey(keyId, algorithmType);
+        if (signingKey == null) {
+            throw new SigningServiceException(String.format("No key for id %s and algorithm %s available.", keyId, algorithmType));
+        }
+        // set the configured kid if present.
+        if (kid.isPresent()) {
+            // we need to clone the key first, to not change the kid of the original key so that the next request still can find it.
+            signingKey = signingKey.cloneKey();
+            signingKey.setKid(keyId);
```

Comment:
The keywrapper generates the keyId on import. However, there are 3 option for the kid in the header for vcs:
- absolute, f.e did:web:my-domain.org#<my-kid>
- just the key: <my-kid>
- in case of did:key, it could also be just did:key:publicKey
Thus, it needs to be possible to overwrite the generated keyId from keycloak.  
Actually, the setter in l65 needs to be removed or the next request wont find the right key anymore(thus, I clone it before setting the new id)

Reply:
fixed in #[b454ed3](https://github.com/keycloak/keycloak/pull/27071/commits/b454ed3f1c975d86e3e8ec05182d82bb2c2c273b)

Reply:
Thank you. I confirmed it.